serde/src/main/java/com/amazon/ionhiveserde/objectinspectors/map/IonFieldNameToBigIntObjectInspector.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amazon.ionhiveserde.objectinspectors.map;
+
+import org.apache.hadoop.hive.serde2.objectinspector.primitive.LongObjectInspector;
+import org.apache.hadoop.hive.serde2.typeinfo.TypeInfoFactory;
+import org.apache.hadoop.io.LongWritable;
+
+public class IonFieldNameToBigIntObjectInspector
+        extends AbstractFieldNameObjectInspector<Long>
+        implements LongObjectInspector {
+
+    public IonFieldNameToBigIntObjectInspector() {
+        super(TypeInfoFactory.longTypeInfo);
+    }
+
+    @Override
+    public long get(final Object o) {
+        return (long) getPrimitiveJavaObject(o);
+    }
+
+    @Override
+    public Object getPrimitiveJavaObject(final Object o) {
+        return getPrimitiveJavaObjectFromFieldName(o.toString());
+    }
+
+    @Override
+    public Object getPrimitiveWritableObject(final Object o) {
+        return new LongWritable(getPrimitiveJavaObjectFromFieldName(o.toString()));
+    }
+
+    @Override
+    protected Long getValidatedPrimitiveJavaObject(final String fieldName) {
+        try {
+            return Long.parseLong(fieldName);
```

Comment:
This question applies to all of the `AbstractFieldNameObjectInspector` implementations.

It looks like the field names are expected to be valid text Ion (they're written as text Ion in `AbstractStructSerializer.getKey`), but here they're read using other Java parsing routines. They should be read as text Ion, which will guarantee that the values are roundtripped equivalently.

Reply:
Thanks for the example! Made some slight changes from this template to get rid of some NPEs, but implemented it as IonPrimitiveReader in utils.